src/frontend/src/lib/utils/token.utils.ts

```diff
@@ -131,3 +139,108 @@ export const mapTokenUi = ({
 		$exchanges
 	})
 });
+
+/**
+ * Type guard to check if a token is of type RequiredTokenWithLinkedData.
+ * This checks whether the token has a twinTokenSymbol field and ensures that it is a string.
+ *
+ * @param token - The token object to be checked.
+ * @returns A boolean indicating whether the token is of type RequiredTokenWithLinkedData.
+ */
+export const isRequiredTokenWithLinkedData = (token: Token): token is RequiredTokenWithLinkedData =>
+	'twinTokenSymbol' in token && typeof token.twinTokenSymbol === 'string';
+
+/**
+ * Type guard to check if an object is of type TokenGroupUi.
+ *
+ * @param tokenUiOrGroupUi - The object to check.
+ * @returns A boolean indicating whether the object is a TokenGroupUi.
+ */
+export const isTokenGroupUi = (tokenUiOrGroupUi: TokenUiOrGroupUi): tokenUiOrGroupUi is TokenGroupUi =>
+	typeof tokenUiOrGroupUi === 'object' &&
+	tokenUiOrGroupUi !== null &&
```

Comment:
`!== null` instead of `isNullish` because TS is not able to  interpret it or because `undefined` is valid?

Reply:
6af63074